google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionManagerImpl.java

```diff
@@ -80,6 +80,13 @@ public void commit() {
     } catch (SpannerException e2) {
       txnState = TransactionState.COMMIT_FAILED;
       throw e2;
+    } finally {
+      // At this point, if the TransactionState is not ABORTED, then the transaction has reached a
+      // terminal state.
```

Comment:
super-nit: I would recommend using `end state` instead of `terminal state`.

Reply:
sure, I have used this term from existing code. But end state is more clear in terms of readability.
https://github.com/googleapis/java-spanner/blob/52915cdb829ed20259ba34e2b69fa163d51cab1a/google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionManager.java#L49